catalog/core/catalog-core-api/src/main/java/ddf/catalog/content/operation/ContentMetadataExtractor.java

```diff
@@ -0,0 +1,39 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p/>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p/>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ **/
+package ddf.catalog.content.operation;
+
+import java.util.Set;
+
+import ddf.catalog.data.AttributeDescriptor;
+import ddf.catalog.data.Metacard;
+
+/**
+ * Parses content, extracting metadata and adding it to the provided card.
+ */
+public interface ContentMetadataExtractor {
```

Comment:
I'd be ok with either. Or both. I can easily imagine scenarios that aren't unstructured text but can still have meaningful metadata parsed out of them. It doesn't give me any heartburn to imagine a text-oriented interface for the common simple case, alongside a more generic content interface that takes input streams. There are some pitfalls to dealing with input streams that it could be nice to avoid for the common case.